tests/scipy_stats_test.py

```diff
@@ -684,6 +685,203 @@ def testMultivariateNormalLogpdfBatch(self, ndim, nbatch, dtype):
+    covariance = L @ L.T
+
+    args_maker = lambda: [
+      rng(shape, dtype), rng(shape[-1:], dtype)]
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker,
+                            tol={np.float32: 1e-3, np.float64: 1e-14})
+    self._CompileAndCheck(
+        lax_fun, args_maker, rtol={np.float32: 3e-07, np.float64: 4e-15})
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": jtu.format_test_name_suffix("", [shape], [dtype]),
+       "dtype": dtype,
+       "shape": shape}
+      for shape in [(15,), (12,)]
+      for dtype in jtu.dtypes.floating))
+  def testKdeIntegrateBox1d(self, shape, dtype):
+    def scipy_fun(dataset, weights):
+      kde = osp_stats.gaussian_kde(dataset, weights=np.abs(weights))
+      # Note: the scipy implementation _always_ returns float64
+      return kde.integrate_box_1d(-0.5, 1.5).astype(dtype)
+
+    def lax_fun(dataset, weights):
+      kde = lsp_stats.gaussian_kde(dataset, weights=jax.numpy.abs(weights))
+      return kde.integrate_box_1d(-0.5, 1.5)
+
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: [
+      rng(shape, dtype), rng(shape[-1:], dtype)]
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker,
+                            tol={np.float32: 1e-3, np.float64: 1e-14})
+    self._CompileAndCheck(
+        lax_fun, args_maker, rtol={np.float32: 3e-07, np.float64: 4e-15})
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": jtu.format_test_name_suffix("", [shape], [dtype]),
+       "dtype": dtype,
+       "shape": shape}
+      for shape in [(15,), (3, 15), (1, 12)]
+      for dtype in jtu.dtypes.floating))
+  def testKdeIntegrateKde(self, shape, dtype):
+    def scipy_fun(dataset, weights):
+      kde = osp_stats.gaussian_kde(dataset, weights=np.abs(weights))
+      other = osp_stats.gaussian_kde(
+        dataset[..., :-3] + 0.1, weights=np.abs(weights[:-3]))
+      # Note: the scipy implementation _always_ returns float64
+      return kde.integrate_kde(other).astype(dtype)
+
+    def lax_fun(dataset, weights):
+      kde = lsp_stats.gaussian_kde(dataset, weights=jax.numpy.abs(weights))
+      other = lsp_stats.gaussian_kde(
+        dataset[..., :-3] + 0.1, weights=jax.numpy.abs(weights[:-3]))
+      return kde.integrate_kde(other)
+
+    rng = jtu.rand_default(self.rng())
+    args_maker = lambda: [
+      rng(shape, dtype), rng(shape[-1:], dtype)]
+    self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker,
+                            tol={np.float32: 1e-3, np.float64: 1e-14})
+    self._CompileAndCheck(
+        lax_fun, args_maker, rtol={np.float32: 3e-07, np.float64: 4e-15})
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": jtu.format_test_name_suffix("", [shape], [dtype]),
+       "dtype": dtype,
+       "shape": shape}
+      for shape in [(15,), (3, 15), (1, 12)]
+      for dtype in jtu.dtypes.floating))
+  @jax.numpy_dtype_promotion("standard")
```

Comment:
When type promotion is strict, this test currently fails, but not because of this code. In fact it looks like it's actually caused by another issue in `random.choice` that I've just opened: https://github.com/google/jax/issues/11285